src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesGraphProjectContextProvider.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.ProjectSystem.VS.Extensibility;
+using Microsoft.VisualStudio.ProjectSystem.VS.Utilities;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    /// <summary>
+    /// Global scope contract that provides information about project level 
+    /// dependencies graph contexts.
+    /// </summary>
+    [Export(typeof(IDependenciesGraphProjectContextProvider))]
+    [AppliesTo(ProjectCapability.AlwaysAvailable)]
+    public class DependenciesGraphProjectContextProvider : IDependenciesGraphProjectContextProvider
+    {
+        private IProjectExportProvider ProjectExportProvider { get; set; }
+
+        private SVsServiceProvider ServiceProvider { get; set; }
```

Comment:
I removed set here to make it readonly, i like properties more than fields since for properties editor suggests who is using them , for fields not

---
In reply to: [71807216](https://github.com/dotnet/roslyn-project-system/pull/309#discussion_r71807216) [](ancestors = 71807216)